_overviews/overview/a-taste-of-scala.md

```diff
@@ -0,0 +1,1329 @@
+
+```scala
+enum CrustSize:
+  case Small, Medium, Large
+
+enum CrustType:
+  case Thin, Thick, Regular
+
+enum Topping:
+  case Cheese, Pepperoni, BlackOlives, GreenOlives, Onions
+```
+
+Once you have an enum you can use it in all of the ways you normally use a trait, class, or object:
+
+```scala
+import CrustSize._
+val currentCrustSize = Small
+
+// enums in a `match` expression
+currentCrustSize match 
+  case Small => println("Small crust size")
+  case Medium => println("Medium crust size")
+  case Large => println("Large crust size")
+
+// enums in an `if` statement
+if currentCrustSize == Small then println("Small crust size")
+```
+
+Enums can also take parameters and have user-defined members like fields and methods. Here’s a sneak-peek of those capabilities:
+
+```scala
+enum Planet(mass: Double, radius: Double):
+  private final val G = 6.67300E-11
+  def surfaceGravity = G * mass / (radius * radius)
+
+  case Mercury extends Planet(3.303e+23, 2.4397e6)
+  case Venus   extends Planet(4.869e+24, 6.0518e6)
+  // more ...
+}
+```
+
+Enums are covered in detail in the Data Modeling section of this Overview, and in the Reference documentation.
+
+
+#### Case classes
+
+A *case class* is an extension of the base Scala class. Case classes provide features that make them useful for functional programming. They have all of the functionality of a regular class, and more. When the compiler sees the `case` keyword in front of a `class`, it generates code for you, with these benefits:
+
+- Case class constructor parameters are public `val` fields by default, so accessor methods are generated for each parameter.
+- An `apply` method is created in the companion object of the class, so you don’t need to use the `new` keyword to create a new instance of the class.
+- An `unapply` method is generated, which lets you use case classes in more ways in `match` expressions.
+- A `copy` method is generated in the class. This provides a way to clone an object while making updates to its values during the cloning process.
+- `equals` and `hashCode` methods are generated.
+- A default `toString` method is generated, which is helpful for debugging.
+
+This code demonstrates several case class features:
+
+```scala
+// define a case class
+case class Person(
+  name: String, 
+  vocation: String
+)
+
+// create an instance of the case class
+val p = Person("Reginald Kenneth Dwight", "Singer")
+
+// a good default toString method
+p                // Person = Person(Reginald Kenneth Dwight,Singer)
+
+// can access its fields, but they are immutable
+p.name           // "Reginald Kenneth Dwight"
+p.name = "Joe"   // error: can’t reassign a val field
+
+// when you need to make a change, use the `copy` method
+// to “update as you copy”
+val p2 = p.copy(name = "Elton John")
+p2               // Person = Person(Elton John,Singer)
+```
+
+See the Data Modeling sections of this Overview and the Reference documentation for many more details on case classes.
+
+
+
```

Comment:
I would expect a mention of sealed traits in the “FP data modeling” section

Reply:
Thanks, I’ll add this.